### PR TITLE
fix: classify 'Request was aborted' as transient and retry unknown erro…

### LINKED ISSUE #293: Testing perspective fails: unknown error class skips retries, aborted requests not classified as transient
## Bug

The Testing perspective fails with exit code 1 across primary + all fallback models, even though Correctness and Security pass on the same primary model (`openrouter/minimax/minimax-m2.5`).

**Run:** https://github.com/misty-step/vox/actions/runs/22501104040
**PR:** https://github.com/misty-step/vox/pull/340
**Cerberus version:** `93b2e9ca` (master)
**pi-version:** 0.55.0

## Symptoms

```
--- stderr ---
Request was aborted.

pi exit=1 stdout=0 bytes scratchpad=0 bytes model=openrouter/minimax/minimax-m2.5 (attempt 1/4)
Unknown error type (exit=1). Trying next model if available...
Model openrouter/minimax/minimax-m2.5 exhausted retries (class=unknown). Trying next fallback...

Falling back to model: openrouter/google/gemini-3-flash-preview (fallback 1/3)
pi exit=1 stdout=0 bytes scratchpad=0 bytes model=openrouter/google/gemini-3-flash-preview (attempt 1/4)
Unknown error type (exit=1). Trying next model if available...
Model openrouter/google/gemini-3-flash-preview exhausted retries (class=unknown). Trying next fallback...

Falling back to model: openrouter/z-ai/glm-5 (fallback 2/3)
pi exit=1 stdout=0 bytes scratchpad=0 bytes model=openrouter/z-ai/glm-5 (attempt 1/4)
Unknown error type (exit=1). Trying next model if available...
```

All 3 models: exit=1, 0 bytes stdout, 0 bytes scratchpad. "Request was aborted" on stderr.

Meanwhile, Correctness and Security passed with the **same primary model** on the same run.

### 1. `class=unknown` skips retries

Each model shows `(attempt 1/4)` then immediately "exhausted retries (class=unknown)." The unknown error class causes immediate retry exhaustion — only 1 attempt instead of the configured 4. Expected: 4 attempts for unknown errors (transient until proven otherwise), or at minimum 2.

### 2. "Request was aborted" not classified as transient

The `pi` stderr contains "Request was aborted" — a timeout or network interruption. This should be classified as a transient/timeout error (alongside rate limits, 429s, 503s) so retries fire. Currently it falls through to "unknown" which then hits bug #1.

## Expected Behavior

- "Request was aborted" → classified as `class=timeout` → retries fire (4 attempts)
- `class=unknown` → at least 2 retry attempts before exhaustion (defensive)
- If all retries fail → SKIP verdict (not FAIL) since no review was produced

## Context

- Correctness: `openrouter/minimax/minimax-m2.5` → PASS (same model, same run)
- Security: `openrouter/minimax/minimax-m2.5` → PASS (same model, same run)
- Testing: 3 models × 1 attempt each → all exit=1 → job fails
- Review runtime: ~7 min (19:42:23 → 19:49:16), within the 600s timeout

### PR DESCRIPTION
**Pull request by moomooskycow**
## Description

This PR fixes a critical reliability issue where "Request was aborted" (network timeout/interruption) was misclassified as an `unknown` error. This caused the reviewer job to skip retries and hard-fail the CI instead of attempting fallbacks or emitting a SKIP verdict.

### Changes

- **`scripts/lib/runtime_facade.py`**: Added `"request was aborted"` to the `transient` (`network`) error classification regex.
- **`scripts/run-reviewer.py`**:
    - Unified transient + unknown retry into single parameterized path (`MAX_UNKNOWN_RETRIES=1`).
    - Unknown errors with output (stdout, stderr, or scratchpad) → SKIP verdict.
    - Unknown errors without any output (infrastructure failure) → job fails.
    - Removed dead fallback code ("Unknown error but output exists — delegating to parser") made unreachable by the unified `has_output` check.
    - Removed unused `shutil` import and trailing whitespace.
- **`.gitignore`**: Added `.pi/settings.json` to gitignore (machine-specific paths).
- **`.pi/agents/worker.md`**: Fixed ambiguous "(unknown)" placeholder in success criteria.

## What Changed

```mermaid
flowchart TD
    A[Pi attempt fails] --> B{classify_runtime_error}
    B -->|transient| C{retry_count < MAX_RETRIES?}
    B -->|unknown| D{retry_count < MAX_UNKNOWN_RETRIES?}
    B -->|permanent| E[Advance to next model]
    C -->|yes| F[Retry with backoff]
    C -->|no| E
    D -->|yes| F
    D -->|no| E
    E --> G{More models?}
    G -->|yes| H[Try next model]
    G -->|no| I{has_output?}
    I -->|"stdout, stderr, or scratchpad"| J["SKIP verdict (exit 0)"]
    I -->|no output| K["Fail job (infrastructure error)"]
    H --> A
    F --> A
    
    style D fill:#90EE90
    style I fill:#90EE90
    style J fill:#90EE90
    style K fill:#FFA07A
```

## Before / After

**Before:** "Request was aborted" → classified as `unknown` → no retries → CI hard-failure. Unknown errors that exhausted all models either hard-failed (pre-PR) or blindly SKIPped everything including infrastructure failures (initial PR version). Retry paths for transient and unknown were structurally duplicated. `has_output` check missed scratchpad, creating inconsistency with legacy fallback.

**After:** "Request was aborted" → `transient/network` → retries up to 3 times. Unknown errors → 1 retry, then model fallback, then output-gated SKIP. Single unified retry path parameterized by error type. `has_output` checks all three output locations (stdout, stderr, scratchpad) consistently. Dead fallback code removed. 1455 tests pass (7 new), lint clean.

### Automated Tests (1455 passed, 1 skipped)
- `test_unknown_error_retries_once` — unknown → retry → success
- `test_request_was_aborted_is_transient` — "Request was aborted" → transient/network → retried
- `test_unknown_error_exhausts_retries` — no output → fail (infrastructure failure)
- `test_unknown_error_with_output_skips` — stderr output → SKIP
- `test_unknown_error_with_scratchpad_only_skips` — scratchpad only → SKIP
- `test_unknown_error_fallback_model_succeeds` — unknown retry exhausted → fallback model succeeds
- `test_request_was_aborted_is_network` — unit test for classifier

Closes #293

## Summary by CodeRabbit

* **New Features**
  * Introduced Pi local workflow with configurable agent and prompt templates for planning, implementation, and review cycles
  * Added standardized agent personas (planner, worker, reviewer) and team role mappings

* **Documentation**
  * Reworked agent manifesto and added comprehensive local workflow and prompt guidance

* **Bug Fixes**
  * Improved runtime error classification and refined retry and unknown-error handling for reviewer runs

* **Refactor**
  * Moved evaluation transform logic to an external module

* **Tests**
  * Expanded runtime tests covering retry, fallback, and error/output edge cases

### CODE REVIEW

File: .pi/settings.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
+  "packages": [],
+  "extensions": [
+    "+/Users/phaedrus/.pi/agent/extensions/organic-workflows",
+    "+/Users/phaedrus/.pi/agent/extensions/profiles",
+    "+/Users/phaedrus/.pi/agent/extensions/subagent",
+    "+/Users/phaedrus/.pi/agent/extensions/orchestration",
+    "+/Users/phaedrus/.pi/agent/extensions/web-search"
```
[the remark this answers is not in the document]
Comment (phrazzld) [reply .pi/settings.json]: Fixed in 8a0284b. .pi/settings.json removed from tracking and added to .gitignore.
